### PR TITLE
[fix-compile-warnings] fix compiler warnings, and add werror in CMakeLists.txt

### PR DESCRIPTION
**Pull request by sethbrin**
fix compiler warnings, and add werror in CMakeLists.txt

### CODE REVIEW

File: src/.ycm_extra_conf.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (sethbrin) [reply src/.ycm_extra_conf.py]: just for the user using tools `YouCompleteMe` to auto complete.

File: src/layer.h [hunk 2]
```diff
@@ -101,6 +101,9 @@ Layer* create_layer(int index);
 #define DEFINE_LAYER_CREATOR(name) \
     ::ncnn::Layer* name##_layer_creator() { return new name; }
 
+#define NCNN_MARK_USED_VAR(var) \
```
[the remark this answers is not in the document]
Comment (sethbrin) [reply src/layer.h]: Ok, I will fix it then.

File: src/layer/arm/convolution_3x3.h [hunk 3]
```diff
@@ -52,10 +51,6 @@ static void conv3x3s1_neon(const Mat& bottom_blob, Mat& top_blob, const Mat& _ke
             const float* r2 = img0 + w*2;
             const float* r3 = img0 + w*3;
 
-            const float* k0 = kernel0;
```
[the remark this answers is not in the document]
Comment (sethbrin) [reply src/layer/arm/convolution_3x3.h]: or just use `static_cast<void>` to mark the variable been used?

[the remark this answers is not in the document]
Comment (nihui) [reply src/layer/arm/convolution_3x3.h]: hi I would prefer move the lines into the following ```#if __ARM_NEON``` else block.
```
#if __ARM_NEON
            float32x4_t _k0123 = vld1q_f32(kernel0);
            float32x4_t _k3456 = vld1q_f32(kernel0+3);
            float32x4_t _k6789 = vld1q_f32(kernel0+6);
#else
            const float* k0 = kernel0;
            const float* k1 = kernel0 + 3;
            const float* k2 = kernel0 + 6;
#endif // __ARM_NEON
```

File: src/layer/arm/convolution_3x3.h [hunk 4]
```diff
@@ -502,6 +497,7 @@ static void conv3x3s1_neon(const Mat& bottom_blob, Mat& top_blob, const Mat& _ke
 
 static void conv3x3s1_winograd64_transform_kernel_neon(const Mat& kernel, Mat& kernel_tm, int inch, int outch)
 {
+    NCNN_MARK_USED_VAR(kernel_tm);
```
[the remark this answers is not in the document]
Comment (sethbrin) [reply src/layer/arm/convolution_3x3.h]: sorry, i didn't notice that.

File: src/layer/arm/convolution_3x3.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (sethbrin) [reply src/layer/arm/convolution_3x3.h]: done.

**Pull request merged**